api/client/client.go

```diff
@@ -1239,3 +1222,61 @@ func (c *Client) DeleteToken(ctx context.Context, name string) error {
 	_, err := c.grpc.DeleteToken(ctx, &types.ResourceRequest{Name: name}, c.callOpts...)
 	return trail.FromGRPC(err)
 }
+
+// GetNodes returns a list of nodes by namespace. If namespace is not specified, the default
+// namespace is used. Nodes that the user doesn't have access to are filtered out.
+func (c *Client) GetNodes(ctx context.Context, namespace string) ([]types.Server, error) {
+	if namespace == "" {
+		namespace = defaults.Namespace
+	}
```

Comment:
DeleteNode/DeleteAllNodes treat empty namespace as an error, but here it's defaulted.
Let's make it an error everywhere.